app/assets/javascripts/oxalis/controller/viewmodes/arbitrary_controller.js

```diff
@@ -223,7 +218,11 @@ class ArbitraryController extends React.PureComponent {
       },
 
       // Recenter active node
-      s: () => this.centerActiveNode(),
+      s: () => {
+        getActiveNode(Store.getState().tracing).map(activeNode =>
+          Store.dispatch(setActiveNodeAction(activeNode.id)),
```

Comment:
Can you change that to `api.tracing.centerPositionAnimated(activeNode.position, false, activeNode.rotation);`? Essentially, you are doing `activeNode = activeNode` and rely on the side effect by the sagas. However, I'd focus on the actual intent which is invoking the side effect.